app/Http/Controllers/CarController.php

```diff
@@ -55,15 +55,24 @@ public function store(Request $request)
         // }
         $data = $request->validate([
             //'key' => 'value'
-            'carTitle' => 'required|string',
+            'carTitle' => 'required|string|regex:/^[A-Za-z]/', //using regex: to force the user that string must start with letter
             'description' => 'required|string|max:1000',
             'price' => 'required|numeric',
+            'image' => 'required|image|mimes:jpeg,png,jpg,gif|max:2048',
             // 'published' => 'boolean',
-        ]);
+        ], ['carTitle.regex' => 'the carTitle field must begin with a letter.']); //Custom Error Message 
 
-         $data['published'] = isset($request->published);
-         dd($data);
+        $data['published'] = isset($request->published);
         
+        $file_extension = $request->image->getClientOriginalExtension();
+        $file_name = time() . '.' . $file_extension;
+        $path = 'assets/images';
+        $request->image->move($path, $file_name);
```

Comment:
do you note you duplicate the code 
```php
        $file_extension = $request->image->getClientOriginalExtension();
        $file_name = time() . '.' . $file_extension;
        $path = 'assets/images';
        $request->image->move($path, $file_name);
```

here is a small tip, when you find yourself repeat the code think of extracting what you repeat inside a function, this is much cleaner

although we will modify it a little bit in the session, but good job ya Omneya